tile.py

```diff
@@ -0,0 +1,205 @@
+        return self._blank
+        
+    def analyze(self):
+        """"Determine dominant colors of target tile, and save that information"""
+        if self.blank:
+            return
+        regions = split_quadrants(self)
+        self.rgb = map(dominant_color, regions) 
+        self.lab = map(cs.rgb2lab, self.rgb)
+
+    def determine_blankness(self, min_depth=1):
+        """Decide whether this tile is blank. Where the mask is grey, tiles
+        and blanked probabilitisically. The kwarg min_depth limits this
+        scattered behavior to small tiles."""
+        if not self._mask: # no mask
+            self._blank = False
+            return
+        brightest_pixel = self._mask.getextrema()[1]
+        if brightest_pixel == 0: # black mask 
+            self._blank = True
+        elif brightest_pixel == 255: # white mask
+            self._blank = False
+        elif self._depth < min_depth: # gray mask -- big tile
+            self._blank = True
+        elif 255*np.random.rand() > brightest_pixel: # small tile
+            self._blank = True
+        else:
+            self._blank = False
+        return
+
+    def straddles_mask_edge(self):
+        """A tile straddles an edge if it contains PURE white (255) and some
+        nonwhite. A tile that contains varying shades of gray does not
+        straddle an edge."""
+        if not self._mask:
+            return False
+        darkest_pixel, brightest_pixel = self._mask.getextrema()
+        if brightest_pixel != 255:
+            return False
+        if darkest_pixel == 255:
+            return False
+        return True
+ 
+    def dynamic_range(self):
+        """What is the dynamic range in this image? Return the
+        average dynamic range over RGB channels. Blur the image
+        first to smooth away outliers."""
+        return sum(map(lambda (x, y): y - x, 
+                       self._img.filter(ImageFilter.BLUR).getextrema()))//3
+
+    def procreate(self):
+        """Divide image into quadrants, make each into a child tile,
+        and return them all in a list.""" 
+        width = self._img.size[0] // 2
+        height = self._img.size[1] // 2
+        children = []
+        for y in [0, 1]:
+            for x in [0, 1]:
+                tile_img = self._img.crop((x*width, y*height,
+                                    (x + 1)*width, (y + 1)*height))
+                if self._mask:
+                    mask_img = self._mask.crop((x*width, y*height,
+                                         (x + 1)*width, (y + 1)*height))
+                else:
+                    mask_img = None
+                child = Tile(tile_img, self.x, self.y,
+                             mask=mask_img,
+                             ancestry=self._ancestry + [(x, y)],
+                             ancestor_size=self._ancestor_size)
+                children.append(child)
+        return children
+        
+    def get_position(self, size, scatter=False, margin=0):
```

Comment:
Scratch that -- this takes too many args to be a property. Let me go make some more coffee...